### PR TITLE
Improve default logs on mounting

### PR DESCRIPTION
**Pull request by gargnitingoogle**
### Description
The following changes in logging.

* Log `File system has been successfully mounted.` on stdout if mount succeeds, even if has specified log-file. This change is only for non-`--foreground` case.
* If log-file is specified, don't log mount-config and flags on stdout, to avoid duplication of log.
* Always prefix all mount-error logs with "Error while mounting gcsfuse: " for clarity.

More detailed log is as follows:

## Case 1 - daemonized run - succcessful mount without log-file
```
$ go run .  $bucket $mountpath
{"time":"28/02/2024 07:46:07.408051","severity":"INFO","message":"Start gcsfuse/unknown (Go version go1.22-20240109-RC01 cl/597041403 +dcbe772469 X:fieldtrack,boringcrypto) for app \"\" using mount point: /usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-mount\n"}
{"time":"28/02/2024 07:46:07.408403","severity":"INFO","message":"GCSFuse mount command flags: {\"AppName\":\"\",\"Foreground\":false,\"ConfigFile\":\"\",\"MountOptions\":{},\"DirMode\":493,\"FileMode\":420,\"Uid\":-1,\"Gid\":-1,\"ImplicitDirs\":false,\"OnlyDir\":\"\",\"RenameDirLimit\":0,\"CustomEndpoint\":null,\"BillingProject\":\"\",\"KeyFile\":\"\",\"TokenUrl\":\"\",\"ReuseTokenFromUrl\":true,\"EgressBandwidthLimitBytesPerSecond\":-1,\"OpRateLimitHz\":-1,\"SequentialReadSizeMb\":200,\"MaxRetrySleep\":30000000000,\"StatCacheCapacity\":12710,\"StatCacheTTL\":60000000000,\"TypeCacheTTL\":60000000000,\"HttpClientTimeout\":0,\"MaxRetryDuration\":-1000000000,\"RetryMultiplier\":2,\"LocalFileCache\":false,\"TempDir\":\"\",\"ClientProtocol\":\"http1\",\"MaxConnsPerHost\":100,\"MaxIdleConnsPerHost\":100,\"EnableNonexistentTypeCache\":false,\"StackdriverExportInterval\":0,\"OtelCollectorAddress\":\"\",\"LogFile\":\"\",\"LogFormat\":\"json\",\"ExperimentalEnableJsonRead\":false,\"DebugFuseErrors\":true,\"DebugFuse\":false,\"DebugFS\":false,\"DebugGCS\":false,\"DebugHTTP\":false,\"DebugInvariants\":false,\"DebugMutex\":false}"}
{"time":"28/02/2024 07:46:07.408492","severity":"INFO","message":"GCSFuse mount config flags: {\"CreateEmptyFile\":false,\"Severity\":\"INFO\",\"Format\":\"json\",\"FilePath\":\"\",\"LogRotateConfig\":{\"MaxFileSizeMB\":512,\"BackupFileCount\":10,\"Compress\":true},\"MaxSizeMB\":-1,\"CacheFileForRangeRead\":false,\"CacheDir\":\"\",\"TtlInSeconds\":-9223372036854775808,\"TypeCacheMaxSizeMB\":4,\"StatCacheMaxSizeMB\":-9223372036854775808,\"EnableManagedFolders\":true}"}
{"time":"28/02/2024 07:46:08.693577","severity":"INFO","message":"File system has been successfully mounted."}
```

## Case 2 - foreground run - succcessful mount without log-file
```
$ go run . --foreground  $bucket $mountpath
{"timestamp":{"seconds":1709149615,"nanos":273996882},"severity":"INFO","message":"Start gcsfuse/unknown (Go version go1.22-20240109-RC01 cl/597041403 +dcbe772469 X:fieldtrack,boringcrypto) for app \"\" using mount point: /usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-mount\n"}
{"timestamp":{"seconds":1709149615,"nanos":274290082},"severity":"INFO","message":"GCSFuse mount command flags: {\"AppName\":\"\",\"Foreground\":true,\"ConfigFile\":\"\",\"MountOptions\":{},\"DirMode\":493,\"FileMode\":420,\"Uid\":-1,\"Gid\":-1,\"ImplicitDirs\":false,\"OnlyDir\":\"\",\"RenameDirLimit\":0,\"CustomEndpoint\":null,\"BillingProject\":\"\",\"KeyFile\":\"\",\"TokenUrl\":\"\",\"ReuseTokenFromUrl\":true,\"EgressBandwidthLimitBytesPerSecond\":-1,\"OpRateLimitHz\":-1,\"SequentialReadSizeMb\":200,\"MaxRetrySleep\":30000000000,\"StatCacheCapacity\":12710,\"StatCacheTTL\":60000000000,\"TypeCacheTTL\":60000000000,\"HttpClientTimeout\":0,\"MaxRetryDuration\":-1000000000,\"RetryMultiplier\":2,\"LocalFileCache\":false,\"TempDir\":\"\",\"ClientProtocol\":\"http1\",\"MaxConnsPerHost\":100,\"MaxIdleConnsPerHost\":100,\"EnableNonexistentTypeCache\":false,\"StackdriverExportInterval\":0,\"OtelCollectorAddress\":\"\",\"LogFile\":\"\",\"LogFormat\":\"json\",\"ExperimentalEnableJsonRead\":false,\"DebugFuseErrors\":true,\"DebugFuse\":false,\"DebugFS\":false,\"DebugGCS\":false,\"DebugHTTP\":false,\"DebugInvariants\":false,\"DebugMutex\":false}"}
{"timestamp":{"seconds":1709149615,"nanos":274347432},"severity":"INFO","message":"GCSFuse mount config flags: {\"CreateEmptyFile\":false,\"Severity\":\"INFO\",\"Format\":\"json\",\"FilePath\":\"\",\"LogRotateConfig\":{\"MaxFileSizeMB\":512,\"BackupFileCount\":10,\"Compress\":true},\"MaxSizeMB\":-1,\"CacheFileForRangeRead\":false,\"CacheDir\":\"\",\"TtlInSeconds\":-9223372036854775808,\"TypeCacheMaxSizeMB\":4,\"StatCacheMaxSizeMB\":-9223372036854775808,\"EnableManagedFolders\":true}"}
{"timestamp":{"seconds":1709149615,"nanos":274362882},"severity":"INFO","message":"Creating Storage handle..."}
{"timestamp":{"seconds":1709149615,"nanos":274372182},"severity":"INFO","message":"UserAgent = gcsfuse/unknown (Go version go1.22-20240109-RC01 cl/597041403 +dcbe772469 X:fieldtrack,boringcrypto) (GPN:gcsfuse) (Cfg:0:0)\n"}
{"timestamp":{"seconds":1709149615,"nanos":274681472},"severity":"INFO","message":"Creating a mount at \"/usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-mount\"\n"}
{"timestamp":{"seconds":1709149615,"nanos":274711512},"severity":"INFO","message":"Creating a new server...\n"}
{"timestamp":{"seconds":1709149615,"nanos":274735882},"severity":"INFO","message":"Set up root directory for bucket gargnitin-fuse-test-bucket1"}
{"timestamp":{"seconds":1709149616,"nanos":538466724},"severity":"INFO","message":"Mounting file system \"gargnitin-fuse-test-bucket1\"..."}
{"timestamp":{"seconds":1709149616,"nanos":541304264},"severity":"INFO","message":"File system has been successfully mounted."}
```

## Case 3 - daemonized run - succcessful mount - with log-file
```
$ go run . --log-format=text --log-file=$logfile  $bucket $mountpath

# On std-out
{"time":"28/02/2024 07:50:41.380816","severity":"INFO","message":"Start gcsfuse/unknown (Go version go1.22-20240109-RC01 cl/597041403 +dcbe772469 X:fieldtrack,boringcrypto) for app \"\" using mount point: /usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-mount\n"}
{"time":"28/02/2024 07:50:42.675989","severity":"INFO","message":"File system has been successfully mounted."}

# In log-file
time="28/02/2024 07:50:41.393954" severity=INFO message="Start gcsfuse/unknown (Go version go1.22-20240109-RC01 cl/597041403 +dcbe772469 X:fieldtrack,boringcrypto) for app \"\" using mount point: /usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-mount\n"
time="28/02/2024 07:50:41.394490" severity=INFO message="GCSFuse mount command flags: {\"AppName\":\"\",\"Foreground\":true,\"ConfigFile\":\"\",\"MountOptions\":{},\"DirMode\":493,\"FileMode\":420,\"Uid\":-1,\"Gid\":-1,\"ImplicitDirs\":false,\"OnlyDir\":\"\",\"RenameDirLimit\":0,\"CustomEndpoint\":null,\"BillingProject\":\"\",\"KeyFile\":\"\",\"TokenUrl\":\"\",\"ReuseTokenFromUrl\":true,\"EgressBandwidthLimitBytesPerSecond\":-1,\"OpRateLimitHz\":-1,\"SequentialReadSizeMb\":200,\"MaxRetrySleep\":30000000000,\"StatCacheCapacity\":12710,\"StatCacheTTL\":60000000000,\"TypeCacheTTL\":60000000000,\"HttpClientTimeout\":0,\"MaxRetryDuration\":-1000000000,\"RetryMultiplier\":2,\"LocalFileCache\":false,\"TempDir\":\"\",\"ClientProtocol\":\"http1\",\"MaxConnsPerHost\":100,\"MaxIdleConnsPerHost\":100,\"EnableNonexistentTypeCache\":false,\"StackdriverExportInterval\":0,\"OtelCollectorAddress\":\"\",\"LogFile\":\"/usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-log.log\",\"LogFormat\":\"text\",\"ExperimentalEnableJsonRead\":false,\"DebugFuseErrors\":true,\"DebugFuse\":false,\"DebugFS\":false,\"DebugGCS\":false,\"DebugHTTP\":false,\"DebugInvariants\":false,\"DebugMutex\":false}"
time="28/02/2024 07:50:41.394589" severity=INFO message="GCSFuse mount config flags: {\"CreateEmptyFile\":false,\"Severity\":\"INFO\",\"Format\":\"text\",\"FilePath\":\"/usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-log.log\",\"LogRotateConfig\":{\"MaxFileSizeMB\":512,\"BackupFileCount\":10,\"Compress\":true},\"MaxSizeMB\":-1,\"CacheFileForRangeRead\":false,\"CacheDir\":\"\",\"TtlInSeconds\":-9223372036854775808,\"TypeCacheMaxSizeMB\":4,\"StatCacheMaxSizeMB\":-9223372036854775808,\"EnableManagedFolders\":true}"
time="28/02/2024 07:50:41.394607" severity=INFO message="Creating Storage handle..."
time="28/02/2024 07:50:41.394622" severity=INFO message="UserAgent = gcsfuse/unknown (Go version go1.22-20240109-RC01 cl/597041403 +dcbe772469 X:fieldtrack,boringcrypto) (GPN:gcsfuse) (Cfg:0:0)\n"
time="28/02/2024 07:50:41.394984" severity=INFO message="Creating a mount at \"/usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-mount\"\n"
time="28/02/2024 07:50:41.395020" severity=INFO message="Creating a new server...\n"
time="28/02/2024 07:50:41.395044" severity=INFO message="Set up root directory for bucket gargnitin-fuse-test-bucket1"
time="28/02/2024 07:50:42.672443" severity=INFO message="Mounting file system \"gargnitin-fuse-test-bucket1\"..."
time="28/02/2024 07:50:42.675509" severity=INFO message="File system has been successfully mounted."
```

## Case 4 - foreground run - succcessful mount - with log-file
```
$ go run . --foreground --log-format=text --log-file=$logfile  $bucket $mountpath

# On stdout

# In log-file
time="28/02/2024 07:52:51.935826" severity=INFO message="Start gcsfuse/unknown (Go version go1.22-20240109-RC01 cl/597041403 +dcbe772469 X:fieldtrack,boringcrypto) for app \"\" using mount point: /usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-mount\n"
time="28/02/2024 07:52:51.936487" severity=INFO message="GCSFuse mount command flags: {\"AppName\":\"\",\"Foreground\":true,\"ConfigFile\":\"\",\"MountOptions\":{},\"DirMode\":493,\"FileMode\":420,\"Uid\":-1,\"Gid\":-1,\"ImplicitDirs\":false,\"OnlyDir\":\"\",\"RenameDirLimit\":0,\"CustomEndpoint\":null,\"BillingProject\":\"\",\"KeyFile\":\"\",\"TokenUrl\":\"\",\"ReuseTokenFromUrl\":true,\"EgressBandwidthLimitBytesPerSecond\":-1,\"OpRateLimitHz\":-1,\"SequentialReadSizeMb\":200,\"MaxRetrySleep\":30000000000,\"StatCacheCapacity\":12710,\"StatCacheTTL\":60000000000,\"TypeCacheTTL\":60000000000,\"HttpClientTimeout\":0,\"MaxRetryDuration\":-1000000000,\"RetryMultiplier\":2,\"LocalFileCache\":false,\"TempDir\":\"\",\"ClientProtocol\":\"http1\",\"MaxConnsPerHost\":100,\"MaxIdleConnsPerHost\":100,\"EnableNonexistentTypeCache\":false,\"StackdriverExportInterval\":0,\"OtelCollectorAddress\":\"\",\"LogFile\":\"/usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-log.log\",\"LogFormat\":\"text\",\"ExperimentalEnableJsonRead\":false,\"DebugFuseErrors\":true,\"DebugFuse\":false,\"DebugFS\":false,\"DebugGCS\":false,\"DebugHTTP\":false,\"DebugInvariants\":false,\"DebugMutex\":false}"
time="28/02/2024 07:52:51.936628" severity=INFO message="GCSFuse mount config flags: {\"CreateEmptyFile\":false,\"Severity\":\"INFO\",\"Format\":\"text\",\"FilePath\":\"/usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-log.log\",\"LogRotateConfig\":{\"MaxFileSizeMB\":512,\"BackupFileCount\":10,\"Compress\":true},\"MaxSizeMB\":-1,\"CacheFileForRangeRead\":false,\"CacheDir\":\"\",\"TtlInSeconds\":-9223372036854775808,\"TypeCacheMaxSizeMB\":4,\"StatCacheMaxSizeMB\":-9223372036854775808,\"EnableManagedFolders\":true}"
time="28/02/2024 07:52:51.936652" severity=INFO message="Creating Storage handle..."
time="28/02/2024 07:52:51.936667" severity=INFO message="UserAgent = gcsfuse/unknown (Go version go1.22-20240109-RC01 cl/597041403 +dcbe772469 X:fieldtrack,boringcrypto) (GPN:gcsfuse) (Cfg:0:0)\n"
time="28/02/2024 07:52:51.937071" severity=INFO message="Creating a mount at \"/usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-mount\"\n"
time="28/02/2024 07:52:51.937119" severity=INFO message="Creating a new server...\n"
time="28/02/2024 07:52:51.937140" severity=INFO message="Set up root directory for bucket gargnitin-fuse-test-bucket1"
time="28/02/2024 07:52:53.194637" severity=INFO message="Mounting file system \"gargnitin-fuse-test-bucket1\"..."
time="28/02/2024 07:52:53.198274" severity=INFO message="File system has been successfully mounted."
```

## Case 5 - daemonized run - failed mount - with log-file
```
$ go run . --log-format=text --log-file=$logfile  a$bucket $mountpath

# On std-out
{"time":"04/03/2024 10:48:56.240802","severity":"INFO","message":"Start gcsfuse/unknown (Go version go1.22-20240109-RC01 cl/597041403 +dcbe772469 X:fieldtrack,boringcrypto) for app \"\" using mount point: /usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-mount\n"}
daemonize.Run: readFromProcess: sub-process: Error while mounting gcsfuse: mountWithArgs: mountWithStorageHandle: fs.NewServer: create file system: SetUpBucket: Error in iterating through objects: storage: bucket doesn't exist
exit status 1

# In log-file
time="04/03/2024 10:48:56.253789" severity=INFO message="Start gcsfuse/unknown (Go version go1.22-20240109-RC01 cl/597041403 +dcbe772469 X:fieldtrack,boringcrypto) for app \"\" using mount point: /usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-mount\n"
time="04/03/2024 10:48:56.254350" severity=INFO message="GCSFuse mount command flags: {\"AppName\":\"\",\"Foreground\":true,\"ConfigFile\":\"\",\"MountOptions\":{},\"DirMode\":493,\"FileMode\":420,\"Uid\":-1,\"Gid\":-1,\"ImplicitDirs\":false,\"OnlyDir\":\"\",\"RenameDirLimit\":0,\"CustomEndpoint\":null,\"BillingProject\":\"\",\"KeyFile\":\"\",\"TokenUrl\":\"\",\"ReuseTokenFromUrl\":true,\"EgressBandwidthLimitBytesPerSecond\":-1,\"OpRateLimitHz\":-1,\"SequentialReadSizeMb\":200,\"MaxRetrySleep\":30000000000,\"StatCacheCapacity\":12710,\"StatCacheTTL\":60000000000,\"TypeCacheTTL\":60000000000,\"HttpClientTimeout\":0,\"MaxRetryDuration\":-1000000000,\"RetryMultiplier\":2,\"LocalFileCache\":false,\"TempDir\":\"\",\"ClientProtocol\":\"http1\",\"MaxConnsPerHost\":100,\"MaxIdleConnsPerHost\":100,\"EnableNonexistentTypeCache\":false,\"StackdriverExportInterval\":0,\"OtelCollectorAddress\":\"\",\"LogFile\":\"/usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-log.log\",\"LogFormat\":\"text\",\"ExperimentalEnableJsonRead\":false,\"DebugFuseErrors\":true,\"DebugFuse\":false,\"DebugFS\":false,\"DebugGCS\":false,\"DebugHTTP\":false,\"DebugInvariants\":false,\"DebugMutex\":false}"
time="04/03/2024 10:48:56.254421" severity=INFO message="GCSFuse mount config flags: {\"CreateEmptyFile\":false,\"Severity\":\"INFO\",\"Format\":\"text\",\"FilePath\":\"/usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-log.log\",\"LogRotateConfig\":{\"MaxFileSizeMB\":512,\"BackupFileCount\":10,\"Compress\":true},\"MaxSizeMB\":-1,\"CacheFileForRangeRead\":false,\"CacheDir\":\"\",\"TtlInSeconds\":-9223372036854775808,\"TypeCacheMaxSizeMB\":4,\"StatCacheMaxSizeMB\":-9223372036854775808,\"EnableManagedFolders\":true}"
time="04/03/2024 10:48:56.254440" severity=INFO message="Creating Storage handle..."
time="04/03/2024 10:48:56.254448" severity=INFO message="UserAgent = gcsfuse/unknown (Go version go1.22-20240109-RC01 cl/597041403 +dcbe772469 X:fieldtrack,boringcrypto) (GPN:gcsfuse) (Cfg:0:0)\n"
time="04/03/2024 10:48:56.254731" severity=INFO message="Creating a mount at \"/usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-mount\"\n"
time="04/03/2024 10:48:56.254783" severity=INFO message="Creating a new server...\n"
time="04/03/2024 10:48:56.254814" severity=INFO message="Set up root directory for bucket agargnitin-fuse-test-bucket1"
time="04/03/2024 10:48:56.991868" severity=ERROR message="Error while mounting gcsfuse: mountWithStorageHandle: fs.NewServer: create file system: SetUpBucket: Error in iterating through objects: storage: bucket doesn't exist\n"
```

## Case 6 - foreground run - failed mount - with log-file
```
$ go run . --foreground --log-format=text --log-file=$logfile  a$bucket $mountpath

# On stdout
Error while mounting gcsfuse: mountWithArgs: mountWithStorageHandle: fs.NewServer: create file system: SetUpBucket: Error in iterating through objects: storage: bucket doesn't exist
exit status 1

# In log-file
time="04/03/2024 10:51:24.429381" severity=INFO message="Start gcsfuse/unknown (Go version go1.22-20240109-RC01 cl/597041403 +dcbe772469 X:f
ieldtrack,boringcrypto) for app \"\" using mount point: /usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/garg
nitin-fuse-test-bucket1-mount\n"
time="04/03/2024 10:51:24.429701" severity=INFO message="GCSFuse mount command flags: {\"AppName\":\"\",\"Foreground\":true,\"ConfigFile\":\
"\",\"MountOptions\":{},\"DirMode\":493,\"FileMode\":420,\"Uid\":-1,\"Gid\":-1,\"ImplicitDirs\":false,\"OnlyDir\":\"\",\"RenameDirLimit\":0,
\"CustomEndpoint\":null,\"BillingProject\":\"\",\"KeyFile\":\"\",\"TokenUrl\":\"\",\"ReuseTokenFromUrl\":true,\"EgressBandwidthLimitBytesPer
Second\":-1,\"OpRateLimitHz\":-1,\"SequentialReadSizeMb\":200,\"MaxRetrySleep\":30000000000,\"StatCacheCapacity\":12710,\"StatCacheTTL\":600
00000000,\"TypeCacheTTL\":60000000000,\"HttpClientTimeout\":0,\"MaxRetryDuration\":-1000000000,\"RetryMultiplier\":2,\"LocalFileCache\":fals
e,\"TempDir\":\"\",\"ClientProtocol\":\"http1\",\"MaxConnsPerHost\":100,\"MaxIdleConnsPerHost\":100,\"EnableNonexistentTypeCache\":false,\"S
tackdriverExportInterval\":0,\"OtelCollectorAddress\":\"\",\"LogFile\":\"/usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/
test_buckets/gargnitin-fuse-test-bucket1-log.log\",\"LogFormat\":\"text\",\"ExperimentalEnableJsonRead\":false,\"DebugFuseErrors\":true,\"De
bugFuse\":false,\"DebugFS\":false,\"DebugGCS\":false,\"DebugHTTP\":false,\"DebugInvariants\":false,\"DebugMutex\":false}"
time="04/03/2024 10:51:24.429804" severity=INFO message="GCSFuse mount config flags: {\"CreateEmptyFile\":false,\"Severity\":\"INFO\",\"Form
at\":\"text\",\"FilePath\":\"/usr/local/google/home/gargnitin/work/cloud/storage/client/gcsfuse/test_buckets/gargnitin-fuse-test-bucket1-log
.log\",\"LogRotateConfig\":{\"MaxFileSizeMB\":512,\"BackupFileCount\":10,\"Compress\":true},\"MaxSizeMB\":-1,\"CacheFileForRangeRead\":false
,\"CacheDir\":\"\",\"TtlInSeconds\":-9223372036854775808,\"TypeCacheMaxSizeMB\":4,\"StatCacheMaxSizeMB\":-9223372036854775808,\"EnableManage
dFolders\":true}"
time="04/03/2024 10:51:24.429823" severity=INFO message="Creating Storage handle..."
time="04/03/2024 10:51:24.429835" severity=INFO message="UserAgent = gcsfuse/unknown (Go version go1.22-20240109-RC01 cl/597041403 +dcbe7724
69 X:fieldtrack,boringcrypto) (GPN:gcsfuse) (Cfg:0:0)\n"
time="04/03/2024 10:51:24.430346" severity=INFO message="Creating a mount at \"/usr/local/google/home/gargnitin/work/cloud/storage/client/gc
sfuse/test_buckets/gargnitin-fuse-test-bucket1-mount\"\n"
time="04/03/2024 10:51:24.430385" severity=INFO message="Creating a new server...\n"
time="04/03/2024 10:51:24.430403" severity=INFO message="Set up root directory for bucket agargnitin-fuse-test-bucket1"
time="04/03/2024 10:51:25.153457" severity=ERROR message="Error while mounting gcsfuse: mountWithStorageHandle: fs.NewServer: create file system: SetUpBucket: Error in iterating through objects: storage: bucket doesn't exist\n"
```

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - Did sanity testing (mounting, unmounting, listing)
2. Unit tests - Ran locally and passed
3. Integration tests - Ran through presubmit

### CODE REVIEW

**Comment by gargnitingoogle** [comment]
Integration tests passed: [log](http://fusion2/ci/kokoro/prod%3Agcsfuse%2Fgcp_ubuntu%2Fpresubmits%2Fperf_tests%2Fpresubmit/activity/8809049c-6b01-4545-af70-4aed517a8271)

File: main.go [hunk 1]
```diff
@@ -356,7 +366,14 @@ func runCLIApp(c *cli.Context) (err error) {
 		mfs, err = mountWithArgs(bucketName, mountPoint, flags, mountConfig)
 
 		if err == nil {
-			logger.Info("File system has been successfully mounted.")
+			// Print the success message in the log-file/stdout depending on what the logger is set to.
+			logger.Info(SuccessfulMountMessage)
+
+			// Print the success message in stdout also, if logger outputs to something other than stdout.
+			if flags.Foreground && mountConfig.FilePath != "" {
```
[the remark this answers is not in the document]
Comment (gargnitingoogle) [reply main.go]: fixed.

File: main.go [hunk 2]
```diff
@@ -239,18 +243,25 @@ func runCLIApp(c *cli.Context) (err error) {
...
-		logger.Warnf("failed to stringify cli flags: %v", err)
-	} else {
-		logger.Infof("GCSFuse mount command flags: %s", flagsStringified)
-	}
+	// Log mount-config and the CLI flags in the log-file.
+	// If there is no log-file, then log these to stdout.
+	// Do not log these in stdout in case of daemonized run
+	// if these are already being logged into a log-file, otherwise
+	// there will be duplicate logs for these in both places (stdout and log-file).
+	if flags.Foreground || mountConfig.FilePath == "" {
```
Comment (ashmeenkaur) [inline main.go:251]: nit: 
```suggestion
	if flags.Foreground || mountConfig.LogConfig.FilePath == "" {
```

File: main.go [hunk 3]
```diff
@@ -356,7 +369,14 @@ func runCLIApp(c *cli.Context) (err error) {
 		mfs, err = mountWithArgs(bucketName, mountPoint, flags, mountConfig)
 
 		if err == nil {
-			logger.Info("File system has been successfully mounted.")
+			// Print the success message in the log-file/stdout depending on what the logger is set to.
+			logger.Info(SuccessfulMountMessage)
+
+			// Print the success message in stdout also, if logger outputs to a log-file.
+			if mountConfig.FilePath != "" {
```
Comment (ashmeenkaur) [inline main.go:376]:
```suggestion
			if mountConfig.LogConfig.FilePath != "" {
```

File: main.go (hunk 2 above)
Comment (gargnitingoogle) [reply main.go]: Done.

File: main.go (hunk 3 above)
Comment (gargnitingoogle) [reply main.go]: Done

**Review by ashmeenkaur — APPROVED** [review]

File: main.go [hunk 4]
```diff
@@ -239,18 +243,25 @@ func runCLIApp(c *cli.Context) (err error) {
...
+		}
 
-	mountConfigStringified, err := util.Stringify(*mountConfig)
-	if err != nil {
-		logger.Warnf("failed to stringify config-file: %v", err)
-	} else {
-		logger.Infof("GCSFuse mount config flags: %s", mountConfigStringified)
+		mountConfigStringified, err := util.Stringify(*mountConfig)
+		if err != nil {
+			logger.Warnf("failed to stringify config-file: %v", err)
```
Comment (ankitaluthra1) [inline main.go:261]: same here, should we add gcsfuse here in statement in error message as well ?

Comment (gargnitingoogle) [reply main.go:261]: @ankitaluthra1  The current change is mostly just refactoring. The change to add `GCSFuse` in logs is needed many other places as well, such as in `SuccessfulMountMessage` and erros. So, let's do it in a separate PR after discussing and finalizing.

Comment (ankitaluthra1) [reply main.go:261]: yes got it..

**Review by ankitaluthra1 — DISMISSED** [review]

**Review by sethiay — APPROVED** [review]

**Pull request merged**